src/remotelibrarymanager.h

```diff
@@ -0,0 +1,32 @@
+#ifndef REMOTELIBRARYMANAGER_H
+#define REMOTELIBRARYMANAGER_H
+
+#include <QObject>
+#include <QtNetwork>
+#include <kiwix/library.h>
+#include <kiwix/manager.h>
+
+class RemoteLibraryManager : public QObject
```

Comment:
I would name it differently.
This class doesn't handle (manage) a remote library. It handle a request to the server to get the result of a new
request.
Something like OpdsRequestManager would be better.

We could even go further. If the class is responsible to handle request to an opds server, we could (should) move the url creation in `ContentManager::updateLibrary` into this class.

---
Or this is really a RemoteLibraryManager, and so it should handle a library. And so to have `ContentManager::updateRemoteLibrary` implemented here.

Reply:
Why do we use macro for the url's creation ? 

https://github.com/kiwix/kiwix-desktop/blob/a005a5cb2580d44a980a316e4a710011dceec659/src/contentmanager.cpp#L223-L225

Reply:
This is :
- 1. to move constants out of the code.
- 2. give some meanings to the constant.

It seems pretty obvious that in `url.setPort(80);`, `80` is the port of the request, and so probably the port of the catalog server.
But `CATALOG_HOST` and `CATALOG_URL`, despite they have the same values, are not the same things.
`CATALOG_HOST` is the host. So, no port or path.
`CATALOG_URL` is the (full) url of the catalog endpoint. It is use by the library to reconstruct absolute urls from relative ones include in the opds stream. It include at least the port. If the port is different from 80, `CATALOG_URL` should include it and be different than `CATALOG_HOST`.

If we directly put the string in the code, we lost the ideas that they are different things.